### PR TITLE
fix bug in derivative of matrix trace

### PR DESCRIPTION
**Pull request by Upabjojr**
#### References to other Issues or PRs

Fixes #15126 

#### Release Notes

* matrices
  * derivative of trace of matrix now returns the correct result.

### CODE REVIEW

File: sympy/matrices/expressions/tests/test_derivatives.py [hunk 1]
```diff
@@ -86,3 +86,106 @@ def test_matrix_derivative_with_inverse():
...
+    ## Second order:
+
+    # Cookbook example 106:
+    expr = Trace(X**2)
+    assert expr.diff(X) == 2*X.T
+
+    # Cookbook example 107:
+    expr = Trace(X**2*B)
+    # TODO: wrong result
+    #assert expr.diff(X) == (X*B + B*X).T
```
Comment (Upabjojr) [inline sympy/matrices/expressions/tests/test_derivatives.py]: @asmeurer a new bug has emerged. I guess we need more extensive testing (i.e. test all rules by the cookbook).

Comment (asmeurer) [reply sympy/matrices/expressions/tests/test_derivatives.py]: OK. Do you plan to fix this here, or should this be merged as is?

Comment (asmeurer) [reply sympy/matrices/expressions/tests/test_derivatives.py]: For reference, in master, this raises `TypeError: Mix of Matrix and Scalar symbols`.

Comment (Upabjojr) [reply sympy/matrices/expressions/tests/test_derivatives.py]: https://github.com/sympy/sympy/issues/15136

**Pull request merged**